src/libLookup/Lookup.cpp

```diff
@@ -2455,6 +2455,43 @@ void Lookup::StartSynchronization() {
   DetachedFunction(1, func);
 }
 
+bool Lookup::GetDSInfoLoop() {
+  unsigned int counter = 0;
+  {
+    lock_guard<mutex> g(m_mediator.m_mutexDSCommittee);
+    if (m_mediator.m_DSCommittee->size() > 0) {
+      LOG_GENERAL(WARNING,
+                  "DS comm already set, make sure you cleaned variables");
+      return false;
+    }
+  }
+
+  while (counter <= FETCH_LOOKUP_MSG_MAX_RETRY) {
+    GetDSInfoFromLookupNodes();
+    unique_lock<mutex> lk(m_mutexDSInfoUpdation);
+    if (cv_dsInfoUpdate.wait_for(lk, chrono::seconds(NEW_NODE_SYNC_INTERVAL)) ==
+        cv_status::timeout) {
+      counter++;
+
+    } else {
+      break;
+    }
```

Comment:
Maybe we don't need to include GetDSInfoFromLookupNodes() inside the waiting loop.
Besides, the counter and cv.wait_for could be merged into the same while loop.

Reply:
After thinking more, we don't need to modify this piece of code.